### PR TITLE
Fix for OFT-64451	 patron unable to click on LJGD-2Q4 in order to see…

### PR DESCRIPTION
**Pull request by thorntonjn**
… her personal details

Defer htmlEncode until after JSON.stringify in order to properly escape quotes within JSON fields.
Reviewed-by: Michael Collins

### CODE REVIEW

**Comment by thorntonjn** [comment]
I worked with @redbugz  and @intervalia Michael Collins on this.  Michael came up with the code that seems to work in all of our cases. However we would like @lambertsteven to make sure we aren't going to break others.   In particular we weren't sure how not doing the htmlDecode on the title attribute might affect others.  

We tested with  < > " ' &  characters and it seemed to all work.

File: assets/js/fsPerson/viewModels/fsPersonVitals.viewModel.js [hunk 1]
```diff
@@ -34,31 +34,32 @@ function fsPersonVitalsViewModel(scope) {
...
+  scope.title = tempHtmlEncode(scope.name) + '\n' + (scope.lifeSpan || '') + (showDot ? ' • ' : '') + (scope.person.id || '');
   if(scope.options.showBirthPlace && scope.person.birthPlace) {
     scope.title += '\n' + scope.person.birthPlace;
   }
-  scope.title = FS.htmlDecode(scope.title);
 
   return scope;
 }
+
+function tempHtmlEncode(inStr) {
```
Comment (lambertsteven) [inline assets/js/fsPerson/viewModels/fsPersonVitals.viewModel.js]: I think fsModules is always used in conjunction with theme-engage (so we'd have the FS object). in which case, we could just remove this function entirely and use `FS.htmlEncode()`. Thoughts?

Comment (thorntonjn) [inline assets/js/fsPerson/viewModels/fsPersonVitals.viewModel.js]: That seems like a good idea.

Comment (intervalia) [inline assets/js/fsPerson/viewModels/fsPersonVitals.viewModel.js]: We don't want to encode everything, just the 4 elements listed.
FS.htmlEncode does more than just those 4.

**Comment by lambertsteven** [comment]
I'm not sure what happened to @intervalia comment. `htmlEncode` does encode more characters, but are there any characters that would show up in a name that we wouldn't want encoded by using `FS.htmlEncode`?

**Comment by redbugz** [comment]
Looks good to me, seems to handle all of the conditions correctly.

**Comment by redbugz** [comment]
Tests pass, I'm going to merge it.

**Comment by redbugz** [comment]
Joel tested it with all of the characters that htmlEncode handles, and it seems to do just fine

**Pull request merged**